SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -146,8 +146,11 @@ bool USpatialActorChannel::IsDynamicArrayHandle(UObject* Object, uint16 Handle)
 	return RepLayout->Cmds[RepLayout->BaseHandleToCmdIndex[Handle - 1].CmdIndex].Type == ERepLayoutCmdType::DynamicArray;
 }
 
-FRepChangeState USpatialActorChannel::CreateInitialRepChangeState(UObject* Object)
+FRepChangeState USpatialActorChannel::CreateInitialRepChangeState(TWeakObjectPtr<UObject> Object)
 {
+	checkf(!Object->IsPendingKill(), TEXT("Attempted to create initial rep change state on an object which is pending kill. This will fail to create a RepLayout: "), Object->GetName());
```

Comment:
When does this happen if ever?

Reply:
This is there to help debugging https://improbableio.atlassian.net/browse/GCS-903 in case it reappears. When looking at the code it seems very unlikely that this could fail and it may be due to an extreme race condition. But Josh was not able to repro it. As we were unable to repro this, the code was added in case this bug is observed again so that we can have more info on what went wrong.